ios/Exponent/Kernel/Core/EXKernel+DeviceInstallUUID.h

```diff
@@ -0,0 +1,16 @@
+// Copyright 2015-present 650 Industries. All rights reserved.
+
+#import "EXKernel.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface EXKernel (DeviceInstallUUID)
+
+/**
+ *  An id that uniquely identifies this installation of Exponent.
+ */
++ (NSString *)deviceInstallUUID;
```

Comment:
This is used only for legacy Notifications, right? If so, let's mention that so we don't use it anymore.

```suggestion
+ (NSString *)deviceInstallUUID __attribute((deprecated("The installation ID API is deprecated and will be removed with legacy Notifications")));
;
```

Reply:
I made it so that `[EXKernel deviceInstallUUID]` is used by `EXDeviceInstallUUIDManager` kernel service which is provided to bindings (subclasses of unimodules used only in ExpoKit):
- `expo-constants`, so that `installationId` in ExpoKit always returns the "common" ExpoKit ID
- `expo-notifications` so that `installationId` in ExpoKit from now on always returns the "common" ExpoKit ID
- versioned constants so that they don't try to create a UUID in `NSUserDefaults` (as was the implementation for SDKs < 40) which would result in ID change for versioned constants (migration removes ID from `NSUserDefaults` to Keychain).

The benefit of overriding unimodules to ensure they're using "common" ExpoKit ID was supposed to be that we would ensure on an "architecture" level that the ID doesn't ever change in managed workflow for any of the modules no matter their bare implementation. I tended to favor this over "loose integration" with which the modules would use the same storage the same way as ExpoKit, but having a copy of the implementation (this is how `EXConstantsService` was ensuring it uses the `deviceInstallUUID` from ExpoKit, it used `NSUserDefaults` with `EXDeviceInstallUUIDKey`; this is also how `EXInstallationIdProvider` was supposed to use the same UUID had it been added to `expotools` list of extra transform rules at https://github.com/expo/expo/blob/95dffabd0058cea1f8d18175aeb5aadef7d1a95b/tools%2Fexpotools%2Fsrc%2Fversioning%2Fios%2Findex.ts#L1149-L1153).

This, however, is unnecessary if we write bare code well carefully (versioning-compatible), so I will change the implementation so that:
- `expo-constants` uses Keychain and includes migration
- `expo-notifications` uses Keychain and includes migration
- legacy notifications use kernel which does not include migration, instead it uses `expo-constants` to fetch the installation ID
- versioned constants use kernel service which uses `expo-constants` to fetch the installation ID.

This way deprecating `installationId` of `expo-constants` is directly related to deprecating versioned constants and legacy notifications which both are/eventually will be deprecated too. `expo-notifications` have their own implementation which won't be deprecated.

I initially didn't want to connect kernel this directly to `expo-constants`, but you're right to say that it won't do any harm. 👍 Thanks!